src/test/java/org/medicmobile/webapp/mobile/OpenSettingsDialogFragmentTest.java

```diff
@@ -0,0 +1,193 @@
+package org.medicmobile.webapp.mobile;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.CALLS_REAL_METHODS;
+import static org.mockito.Mockito.RETURNS_SMART_NULLS;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.withSettings;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.view.MotionEvent;
+import android.view.View;
+import android.view.View.OnTouchListener;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.MockSettings;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(sdk=28)
+public class OpenSettingsDialogFragmentTest {
+
+	private OpenSettingsDialogFragment openSettingsDialogFragment;
+	private Activity activity;
+	private ArgumentCaptor<OnTouchListener> argsOnTouch;
+	private ArgumentCaptor<Intent> argsStartActivity;
+
+	@Before
+	public void setup() {
+		activity = mock(Activity.class, RETURNS_SMART_NULLS);
+		doNothing().when(activity).finish();
+
+		View view = mock(View.class);
+		argsOnTouch = ArgumentCaptor.forClass(OnTouchListener.class);
+		doNothing().when(view).setOnTouchListener(argsOnTouch.capture());
+
+		MockSettings fragmentSettings = withSettings()
+			.useConstructor(view)
+			.defaultAnswer(CALLS_REAL_METHODS);
+
+		openSettingsDialogFragment = mock(OpenSettingsDialogFragment.class, fragmentSettings);
+		when(openSettingsDialogFragment.getActivity()).thenReturn(activity);
+		argsStartActivity = ArgumentCaptor.forClass(Intent.class);
+		doNothing().when(openSettingsDialogFragment).startActivity(argsStartActivity.capture());
+
+		openSettingsDialogFragment.onCreate(null);
+	}
+
+	private void tap(OnTouchListener onTouchListener, MotionEvent eventTab, int times) {
+		for (int i = 0; i < times; i++) {
```

Comment:
I recently learned about the `IntStream.range` method which is essentially an easy way to created a sized Stream.

That would allow for something like: `IntStream.range(0, times).forEach(i -> onTouchListener.onTouch(null, eventTab));` which is handy if you like functional code and want to avoid loops....  (Though, I don't think it really helps to improve the "readability" of the code...)

Reply:
Nice! I like functional